src/sap/fhir/model/r4/lib/FHIRRequestor.js

```diff
@@ -228,25 +229,40 @@ sap.ui.define([
 	FHIRRequestor.prototype._sendRequest = function(sMethod, sPath, mParameters, mHeaders, oPayload, fnSuccess, fnError, oBinding) {
 		var oRequestHandle = new RequestHandle(oBinding);
 		var oFHIRUrl = new FHIRUrl(sPath, this._sServiceUrl);
-		var sQueryParameter =  oFHIRUrl.getQueryParameters() ? "" : this._buildQueryParameters(mParameters, this.oModel.getBindingInfo(oFHIRUrl.getRelativeUrlWithoutQueryParameters()), sMethod);
-		var sRequestUrl = sPath.startsWith("http") ? sPath : this._sServiceUrl + oFHIRUrl.getRelativeUrlWithQueryParameters() + sQueryParameter;
+		var sRequestUrl;
+		var sContentType = "application/json";
+		var oBindingInfo = this.oModel.getBindingInfo(oFHIRUrl.getRelativeUrlWithoutQueryParameters());
+		// if the method is GET , secure search is enabled
+		// convert the path to _search and all the url paramters will be converted to POST form data
+	    // except ValueSet/$expand or specific operations  like Patient/53/_history
+		if (sMethod == HTTPMethod.GET && this.bSecureSearch && oFHIRUrl.isSearchAtBaseLevel() && !this._isCsrfTokenRequest()) {
```

Comment:
This condition is quite complex and hard to understand. Could we create a private method with a meaningful name and put this condition there?

Reply:
done..moved to a new method `_canRequestBeTransformed`